intake/catalog/local.py

```diff
@@ -584,7 +584,15 @@ def _load(self):
             self._entries[entry.name] = entry
 
         self.metadata = cfg.get('metadata', {})
+        self.name = self.name or cfg.get('name') or self.name_from_path
+        self.description = self.description or cfg.get('description')
 
+    @property
+    def name_from_path(self):
+        name = os.path.splitext(os.path.basename(self.path))[0]
+        if name == 'catalog':
```

Comment:
for "catalog.yaml"? Maybe this should be documented somewhere. It could also be an option to always take the dir name.

Reply:
Yeah for catalog.yaml or catalog.yml. I'll make a docstring and think about where to add it in the docs.

Reply:
@martindurant where to add in the docs?

Reply:
In class docstring, since this is basically something that is decided by the inputs to `__init__`